### PR TITLE
[FIX] mail: wrap chatter content inside div

### PR DESCRIPTION
**Pull request by Mtaylorr**
Steps to reproduce the issue:
=============================
- Go to chatter
- Write content on 2 lines
- Open composer
- Select first line and change the heading
- Both lines are changed

Origin of the issue:
====================
The content of the textarea sent to the server is `abc<br>efg`. 

In 17.4: We don't sanitize the content so it stays the same and we put that content into the editor. The editor will create the `p` elements to wrap the inline elements at root.

In 18.0: We sanitize the content and it will pass by `lxml.html.fromstring` and it will wrap the content in a `p` element. Now we put `<p> abc<br> efg</p>` in the editor which is valid but it's considered all as the same block. Applying block-level commands like list and heading will change the whole content.

Solution:
=========
Wrap the textarea content inside a div when we open the mail composer.

opw-4350430
opw-4346348

### CODE REVIEW

File: addons/mail/static/src/core/common/composer.js [hunk 1]
```diff
@@ -534,9 +534,17 @@ export class Composer extends Component {
...
             mentionedPartners: this.props.composer.mentionedPartners,
         });
+        let default_body = await prettifyMessageContent(body, validMentions);
+        // We need to wrap the content inside a div element before sending the content to the editor
+        // because the textarea inline content will be changed by sanitize on the onchange call
+        // and it will be wrapped inside a p because of html.fromstring in python.
+        // The wrapping inside a p will make all the content as one single block. Applying block-level
+        // commands like (headings, list..) will not work as they will be applied on the top p element
+        // which means the whole content.
+        default_body = "<div>" + default_body + "</div>";
```
Comment (alexkuhn) [inline addons/mail/static/src/core/common/composer.js:544]: Keep comment as brief as you can, without referring to technical details that are not in immediate scope of these LOCs:

- when comments are too long, we don't want to read them, so they just pollute code and we need to scroll more to see whole context of effective code
- when comment refers to how / why with code of now, they will most likely lie in the future and this becomes even harmful. Comments are poor for maintainability, so they should be rare or very brief to give a good grasp of immediate intent here

Here's my suggestion of comment:

```js
default = "<div>" + default_body + "</div>"; // as to not wrap in <p> by html_sanitize
```

The comment is brief and gives a good grasp of the intent. It doesn't tell everything in details, the test coverage is there for more details. Most people will have enough clue of the intent and why this is relevant (or maybe not), with minimal footprint on the code.

_IMO I like inline comments, I feel they are underrated: it forces you to put comment on the actual relevant LOC and keep it as brief as possible (even can make you reconsider rewriting the code differently to not "require" any comment)._

File: addons/mail/static/tests/tours/mail_composer_test_tour.js [hunk 2]
```diff
@@ -150,5 +150,34 @@ registry.category("web_tour.tours").add("mail/static/tests/tours/mail_composer_t
...
+        },
+        {
+            content: "write second line",
+            trigger: ".o-mail-Composer-input",
+            run: "fill efg",
+        },
+        {
+            content: "Open full composer",
+            trigger: "button[aria-label='Full composer']",
+            run: "click",
```
Comment (alexkuhn) [inline addons/mail/static/tests/tours/mail_composer_test_tour.js:172]: Note for forward-port: the `aria-label` has been replaced by `title`, and value has been changed to "Open Full Composer".

File: addons/mail/static/src/core/common/composer.js (hunk 1 above)
Comment (Mtaylorr) [reply addons/mail/static/src/core/common/composer.js]: @alexkuhn , thanks for the review, I updated the fix now as suggested

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**